agent/engine/docker_container_engine.go

```diff
@@ -571,10 +571,12 @@ func (dg *dockerGoClient) ContainerEvents(ctx context.Context) (<-chan DockerCon
 
 	go func() {
 		for event := range events {
-			containerId := event.ID
-			if containerId == "" {
+			// currently only container events type needs to be handled
+			if event.Type != "container" || event.ID == "" {
```

Comment:
Event types were added in Docker 1.10.  Can you verify whether this works with previous versions of Docker (like 1.9)?  We may need to handle the case where `event.Type` is `""`.